### PR TITLE
[FIX] Windows warning: EPR-dictionary: shift negative or too big

### LINKED ISSUE #1908: EPR changes create new warnings on windows
Please double-check the cdash yourself after larger PRs have been merged!

### PR DESCRIPTION
**Pull request by cpockrandt**
(hopefully) resolves #1908 

Thanks for hinting at it, h-2!

### CODE REVIEW

File: include/seqan/index/index_fm_rank_dictionary_levels.h [hunk 1]
```diff
@@ -505,7 +505,8 @@ struct RankDictionary<TValue, Levels<TSpec, TConfig> >
         // TODO(cpockrandt): make all bitmasks a const-expr
         const unsigned maxValue = (1 << _BITS_PER_VALUE) - 1;
         const unsigned padding = _BITS_PER_WORD % _BITS_PER_VALUE;
```
Comment (h-2) [inline include/seqan/index/index_fm_rank_dictionary_levels.h]: don't use the max metafunction, use the stl's max function instead and use it up here. You could also make it constexpr

**Comment by h-2** [comment]
Out of curiosity, have you tried making paddingWord constexpr as well and whether the warning is still created, when padding is constexpr when they all are constexpr? Because then the compiler should know all combinations at compile time....

**Comment by cpockrandt** [comment]
I tried making ``paddingWord`` constexpr but it didn't work because of ``std::max`` I guess (see commit ``f9755d6``)

**Comment by h-2** [comment]
std::max should be constexpr since c++14 so it should work, but I think it should even work without, because the windows compiler know at compile time that the value is >= 1. Can you try either way?

**Comment by h-2** [comment]
@ping

**Comment by cpockrandt** [comment]
@h-2 Introducing ``constexpr`` without ``max()`` does not work on Windows

**Comment by h-2** [comment]
@cpockrandt why does windows suck so much :disappointed: 

Did you introduce make the original solution? then I will merge it...

**Comment by cpockrandt** [comment]
@h-2 As discussed in person, this is the final version, so you're free to merge it :)

**Pull request merged**